digests.go

```diff
@@ -13,23 +13,23 @@ type Digester interface {
 	Digest(io.Reader) (digest.Digest, error)
 }
 
-// ContentProvider produces a read stream for a given digest
+// ContentProvider provides content for a digest.
 type ContentProvider interface {
+	// Reader returns a stream of the content for
+	// the provided digest.
 	Reader(digest.Digest) (io.ReadCloser, error)
+
+	// CopyTo copies the content from the provided digest
+	// to the destination file path.
+	CopyTo(string, digest.Digest) error
```

Comment:
This is an odd interface for providers to implement. Couldn't we hide this behind the `io.ReadCloser` by returning back an `os.File` directly?

Do you have an example implementation?